Change regex to allow commented out -name lines

bookshelfdave opened the pull request:
The regex should ignore vm.args lines like the following:
# -name riak@127.0.0.1
# -name test-riak@node.hode
# use the -name switch to specify the name of the node

-administrator-name admin
-sssssssssname foo

Commit:
Change regex to allow commented out -name lines

Please ignore previous 2 commits to dp_gh169 - I was unsure if my
rebase/squash worked correctly.

jaredmorrow commented:
This does not work on Solaris 10, I will investigate how to change it to work properly.

jaredmorrow committed:
Fix regular expression to work on all platforms Riak supports

Tested on FreeBSD / Ubuntu / Centos / OSX / Solaris
with the following -name settings:

```erlang
-name dev2@127.0.0.1
-administrator-name admin
-sssssssssname foo
    -name dev2@127.0.0.1
-sname dev2
     -sname dev2
```

Uncommenting out each one in sequence

Fixes #175
